client/landing/gutenboarding/onboarding-block/style-preview/disableNavigation.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Changes href attributes' values to "#!" and removes meta-redirect tags.
+ *
+ * @param contentDocument the document you want to process
+ */
+export function disarmAnchorsAndMetaRefreshes( contentDocument: HTMLDocument ) {
+	// dis-arm anchors
+	contentDocument.body
+		?.querySelectorAll( 'a[href]:not([href="#!"])' )
+		.forEach( ( anchor: Element ) => {
+			// we don't want to remove href attribute because styling can change without it
+			// using only # will jump to top
+			anchor.setAttribute( 'href', '#!' );
+		} );
+
+	// remove meta redirects
+	contentDocument.head
+		?.querySelectorAll( 'meta[http-equiv="refresh"]' )
+		.forEach( ( element: Element ) => {
+			element.parentNode?.removeChild( element ); // element.remove() doesn't work in IE
+		} );
+}
+
+/**
+ * Handles all submit events in hte iframe and prevents their default behavior.
+ * Also it handles `load` event for the iframe and calls {@link disarmAnchorsAndMetaRefreshes} on the iframe document when it fires
+ *
+ * @param iframe The iframe you want to process
+ */
+export function disableNavigation( iframe: HTMLIFrameElement ) {
+	const { contentDocument, contentWindow } = iframe;
+
+	if ( contentDocument && contentWindow ) {
+		// submit event bubbles up, we only need to catch it on the window level
```

Comment:
> Is the event listener here still necessary?

Yes, because IE doesn't support [`form-action` directive.](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/form-action)

Reply:
> Are we not concerned about JS intercepting forms in that case (hence stop from bubbling up)?

Yes. That too.

Reply:
> Yes, because IE doesn't support form-action directive.

We could add a comment here maybe (so anyone looking at this knows why it's here). Something like "_we enforce a none form-action CSP, but need to also block here explicitly for IE support (IE does not support form-action directive [link]_"

p.s. it sounds like we don't need the `form-action` CSP. 🤔

Reply:
> p.s. it sounds like we don't need the form-action CSP. 🤔

We do need it some JS intercepts the submit events before it bubbles up to `window`.

Reply:
Or if someone calls `submit()` on the form. See: https://stackoverflow.com/questions/18774821/submit-event-does-not-fire-if-submit-initiated-programmatically

Reply:
> We do need it some JS intercepts the submit event before it bubbles up to window.

Ok. If this is important (JS interception blocking the event), then it would fail in IE in that case right (since it doesn't support the directive). I think you need to clarify these couple of points in the code/comments. It looks a little ambiguous to me i.e. things are not what they seem. 🤔

Reply:
Thanks for adding in the comment above btw! :-)